src/commands/tasks/taskConfig.ts

```diff
@@ -0,0 +1,182 @@
+  if (!config.name) { throw new Error('Task name not specified'); }
+  if (!config.entryPoint) { throw new Error('Entrypoint not specified'); }
+  if (!config.runtime) { throw new Error('Runtime not specified'); }
+
+  if (config.timeLimit !== undefined && (config.timeLimit < limits.time.min || config.timeLimit > limits.time.max)) {
+    throw new Error('ERROR: timeLimit out of bounds!');
+  }
+  if (config.memoryLimit !== undefined && (config.memoryLimit < limits.memory.min || config.memoryLimit > limits.memory.max)) {
+    throw new Error('ERROR: memoryLimit out of bounds!');
+  }
+
+  if (!/^[A-Za-z0-9\-]+$/g.test(config.name)) { throw new Error('Please use only alphanumeric characters for your task name'); }
+  if (!runtimeChoices.includes(config.runtime)) {
+    throw new Error(`Runtime should be one of ${runtimeChoices.join(', ')}`);
+  }
+
+  if (config.path) {
+    try {
+      await fs.access(config.path);
+    } catch (err) {
+      throw new Error(`Please provide a valid directory path for your code, ${config.path} not found`);
+    }
+    if ((await fs.stat(config.path)).isFile()) {
+      throw new Error(`please provide a valid directory path for your code, ${config.path} points to a file`);
+    }
+  } else {
+    throw new Error('Code path not specified');
+  }
+
+  if (config.executionPermission && !Object.values(permissionModes).includes(config.executionPermission)) {
+    throw new Error(`executionPermission incorrect. Should be one of ${Object.values(permissionModes).join(',')}`);
+  }
+  return true;
+}
+
+export async function loadSingleConfigFile(path:string): Promise<TaskConfig> {
+  let taskConfig: TaskConfig;
+  try {
+    /* load config file */
+    const data = await fs.readFile(path);
+    try {
+      taskConfig = JSON.parse(data.toString()) as TaskConfig;
+    } catch (err) {
+      throw new Error(`failed to parse file ${err.message}`);
+    }
+  } catch (err2) {
+    throw new Error(`Invalid config file: ${err2.message}`);
+  }
+
+  /* Resolve variables before doing any further processing */
+  taskConfig = replaceConfigVariables(taskConfig);
+
+  /* The config file path is relative to the location of the config file, so change it */
+  if (taskConfig.path) {
+    taskConfig.path = ospath.join(ospath.dirname(path), taskConfig.path);
+  }
+
+  /* For things which aren't mandatory, set sensible defaults anyway */
+  taskConfig.timeLimit ??= 60;
+  taskConfig.memoryLimit ??= 128;
+  taskConfig.environment ??= {};
+
+  /* Convert environment variables to the value format that the service expects */
+  if (taskConfig.environment) {
+    taskConfig.environment =
+      Object.entries(taskConfig.environment).reduce((prev:any, curr:any) => ({ ...prev, [curr[0]]: { value: curr[1] } }), {});
+  }
+  return taskConfig;
+}
+
+export async function* getValidatedConfigIterator(
+  { path, name, code, entryPoint, runtime, description, timeLimit, memoryLimit, executionPermission, env }: any
```

Comment:
Why not type the function parameters rather than using `any`?

Reply:
I could, but there's not really a point to it. These values come straight from the handler and pass in all stuff from the command line. There's little guarantee what is there and what type it is. The validation occurs in the function itself.
If anything, setting it to _any_ is a warning sign.